core/src/core/transaction.rs

```diff
@@ -272,20 +352,16 @@ impl TxKernel {
 	}
 
 	/// The msg signed as part of the tx kernel.
-	/// Consists of the fee and the lock_height.
+	/// Based on kernel features and associated fields (fee and lock_height).
 	pub fn msg_to_sign(&self) -> Result<secp::Message, Error> {
-		let msg = kernel_sig_msg(self.fee, self.lock_height, self.features)?;
+		let msg = self.features.kernel_sig_msg()?;
 		Ok(msg)
 	}
 
 	/// Verify the transaction proof validity. Entails handling the commitment
 	/// as a public key and checking the signature verifies with the fee as
 	/// message.
 	pub fn verify(&self) -> Result<(), Error> {
-		if self.is_coinbase() && self.fee != 0 || !self.is_height_locked() && self.lock_height != 0
-		{
-			return Err(Error::InvalidKernelFeatures);
-		}
```

Comment:
Does this mean now the TxKernel can have a `fee`>0 for coinbase? and `lock_height`=0 with KernelFeatures=HeightLocked?

because you removed this checking on `verify()`.

Reply:
No. A coinbase kernel has no fee at all, there is no field for it any more.
And anything other than a height locked kernel cannot have a lock height value.

Reply:
I mean if now we broadcast a new block with a coinbase kernel like the following (took part of data from your example):
```sh
01 | 00 00 00 00 01 f7 8a 40 | 00 00 00 00 00 00 00 00 | 08 b1 ... 22 d8 | 33 11 ... b9 69
```

Will it split the Grin network into 2 parts? the 1st part with all old version nodes which reject this block because it has a `fee>0` for coinbase kernel, and the 2nd part with all new version nodes which accept this block?